WasatchNET/EEPROM.cs

```diff
@@ -1928,33 +1934,77 @@ public bool hasLaserPowerCalibration()
 
         void enforceReasonableDefaults()
         {
+            ////////////////////////////////////////////////////////////////////
+            // wavecal (only check first 4)
+            ////////////////////////////////////////////////////////////////////
+
             bool defaultWavecal = false;
             for (int i = 0; i < 4; i++)
                 if (Double.IsNaN(wavecalCoeffs[i]))
                     defaultWavecal = true;
 
-            for (int i = 0; i < 5; i++)
-                if (Double.IsNaN(linearityCoeffs[i]))
-                    linearityCoeffs[i] = 0;
-            for (int i = 0; i < 4; i++)
-                if (Double.IsNaN(laserPowerCoeffs[i]))
-                    laserPowerCoeffs[i] = 0;
-
             if (defaultWavecal || format == 0xff)
             {
                 logger.error("EEPROM appears to be default");
                 defaultValues = true;
+                wavecalCoeffs = new float[5];
                 wavecalCoeffs[0] = 0;
                 wavecalCoeffs[1] = 1;
                 wavecalCoeffs[2] = 0;
                 wavecalCoeffs[3] = 0;
                 wavecalCoeffs[4] = 0;
             }
 
+            // if there's a 5th element, make double-sure it's not corrupt
+            if (wavecalCoeffs.Length == 5)
+            {
+                double t = wavecalCoeffs[wavecalCoeffs.Length - 1];
+                if (double.IsNaN(t) || double.IsInfinity(t))
+                    wavecalCoeffs[wavecalCoeffs.Length - 1] = 0;
+            }
+
+            ////////////////////////////////////////////////////////////////////
+            // other coeffs
+            ////////////////////////////////////////////////////////////////////
+
+            linearityCoeffs = Util.cleanNan(linearityCoeffs);
+            laserPowerCoeffs = Util.cleanNan(laserPowerCoeffs);
+
+            ////////////////////////////////////////////////////////////////////
+            // integration time
+            ////////////////////////////////////////////////////////////////////
+
             if (minIntegrationTimeMS < 1 || minIntegrationTimeMS > 1000)
             {
                 logger.error("invalid minIntegrationTimeMS found ({0}), defaulting to 1", minIntegrationTimeMS);
                 minIntegrationTimeMS = 1;
+                maxIntegrationTimeMS = 60000;
+            }
+
+            ////////////////////////////////////////////////////////////////////
+            // other min/max
+            ////////////////////////////////////////////////////////////////////
+
+            // you'd think this could be done with generics but no...
+            if (minIntegrationTimeMS > maxIntegrationTimeMS)

```

Comment:
I might change these all to >=

Reply:
Though based on the code...idk

Reply:
If the conditional is true, it swaps the values, so wouldn't do much if equal :-)